src/mscorlib/src/Internal/Runtime/CompilerServices/Unsafe.cs

```diff
@@ -10,21 +12,22 @@
 using nuint = System.UInt32;
 #endif
 
-namespace System.Runtime.CompilerServices
+namespace Internal.Runtime.CompilerServices
 {
     //
     // Subsetted clone of System.Runtime.CompilerServices.Unsafe for internal runtime use.
     // Keep in sync with https://github.com/dotnet/corefx/tree/master/src/System.Runtime.CompilerServices.Unsafe.
     // 
 
     /// <summary>
-    /// Contains generic, low-level functionality for manipulating pointers.
+    /// For internal use only. Contains generic, low-level functionality for manipulating pointers. 
     /// </summary>
-    internal static unsafe class Unsafe
+    public static unsafe class Unsafe
```

Comment:
@jkotas I'm surprised to see it go this direction. I would have expected it to already be public and we instead make it internal. Do we know if there are libraries in corefx that depend on corelib and need the Unsafe APIs?

Reply:
> Do we know if there are libraries in corefx that depend on corelib and need the Unsafe APIs?

Yes, there are. System.Memory is the primary one.

Before this change:
- The inbox System.Memory gets Unsafe APIs from System.Runtime.CompilerServices.Unsafe. It pulls System.Runtime.CompilerServices.Unsafe inbox and turns it into partial OOB. It breaks .NET Core 2.0->2.1 roll forward among other things.

After this change:
- The inbox System.Memory gets Unsafe APIs from CoreLib. System.Runtime.CompilerServices.Unsafe can stay as a pure independent nuget package. .NET Core 2.0->2.1 roll forward works fine.

Do you see a problem with this plan?

Reply:
>Yes, there are. System.Memory is the primary one.

I would expect the there are two versions of System.Memory. One which is the OOB which would use the OOB Unsafe library and the other which is the a facade and just type-forwards to CoreLib. Are there types in System.Memory which references CoreLib that still needs these Unsafe methods? If so should they move into CoreLib as well to avoid making this public? 

If there are other APIs that build against CoreLib that use Unsafe that we don't want to move to CoreLib I think this plan should work and I can live with it.